Added migrations for Tips & Donations' settings

sagzy opened the pull request:
refs https://github.com/TryGhost/Product/issues/3668

- `Tips and Donations` feature offers two settings: `donations_currency`, and `donations_suggested_amount`
   - `donation_currency`: the currency to be used for the donation. Defaults to `USD`, not nullable.
   - `donation_suggested_amount`: an anchor price for the donation. Defaults to `0`, not nullable.
- Both settings belong to a new group `donations`

cf. [Tech Spec](https://www.notion.so/ghost/Tech-Spec-5cd6929f7960462ebcbf198176e0d899?pvs=4#6e8b34c45f0c4c78b48c9e7725a307c8)

sagzy committed:
Added migrations for donation settings

refs https://github.com/TryGhost/Product/issues/3668

- "Tips and Donations" feature offers three settings: "donations_enabled", "donations_currency", and "donations_suggested_amount"
    - donation_enabled: whether the "Tips and Donations" feature is enabled or not. Note that this setting is created with a PUBLIC flag, so that it's available in the Settings endpoint of the Contents API
    - donation_currency: the currency to be used for the donation
    - donation_suggested_amount: an anchor price for the donation
- All three settings belong to a new group, "donations"

sagzy commented on ghost/core/test/e2e-api/admin/__snapshots__/settings.test.js.snap:
```diff
@@ -694,6 +706,18 @@ Object {
+    },
+    Object {
+      "key": "donations_suggested_amount",
+      "value": "0",
```
Unsure why snapshots have `"0"` (string) instead of `0` (number) for `donation_suggested_amount`

allouis replied:
We don't cast values to the number type, not sure why https://github.com/TryGhost/Ghost/blob/81ef2ade39e57c8431585d971ba6c601278441a3/ghost/core/core/server/models/settings.js#L139-L160

allouis reviewed (approved):
We don't validate currencies for Tiers either ATM. I think this is all good!

ErisDS reviewed (approved):
LGTM 👍

sagzy committed:
Removed donations_enabled settings

refs https://github.com/TryGhost/Product/issues/3668
- following a discussion with peter, we decided to not introduced a toggle for donations on/off. Instead, donations will be automatically enabled if stripe is connected, disabled otherwise
